lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -425,27 +419,29 @@ public TokenDetails authorize(TokenParams params, AuthOptions options) throws Ab
 		options = (options == null) ? this.authOptions : options.copy();
 		params = (params == null) ? this.tokenParams : params.copy();
 
+		tokenAuth.clear();
```

Comment:
I think this could lead to be bugs, unless I have misunderstood. Why we are forcibly clearing the token before the new one is created? Should we not simply wait for `ably.onAuthUpdated` to clear the token so that whilst `tokenAuth.authorize` is doing it's thing, the client library can still use the existing token?